instance/tests/api/test_openedx_instance.py

```diff
@@ -280,3 +284,229 @@ def test_get_details_unhealthy(self, mock_consul):
+
+    @patch_services
+    def test_instance_list_filter_on_status(self, mock_consul, mock_patch_services):
+        """
+        GET - instance list - it should be possible to filter on any of the
+        app server statuses
+        """
+        instance = OpenEdXInstanceFactory(sub_domain='test.com', name='test.com')
+        instance.ref.owner = self.organization2
+        instance.ref.save()
+
+        instance.spawn_appserver()  # Server state is Running
+
+        instance2 = OpenEdXInstanceFactory(sub_domain='test2.com', name='test2.com')
+        instance2.ref.owner = self.organization2
+        instance2.ref.save()
+
+        instance2.spawn_appserver()
+        instance2.appserver_set.update(_status=AppServerStatus.New.state_id)
+
+        self.api_client.login(username='user3', password='pass')
+        response = self.api_client.get('/api/v1/instance/?status=running')
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['id'], instance.ref.id)
+
+    @patch_services
+    def test_instance_list_filter_on_lifecycle_production(self, mock_consul, mock_patch_services):
+        """
+        GET - instance list - filter on all production instances.
+        """
+        instance = OpenEdXInstanceFactory(
+            sub_domain='test.com',
+            name='test.com',
+            additional_monitoring_emails=['urgent@opencraft.com']
+        )
+        instance.ref.owner = self.organization2
+        instance.ref.save()
+
+        instance2 = OpenEdXInstanceFactory(sub_domain='test2.com', name='test2.com')
+        instance2.ref.owner = self.organization2
+        instance2.ref.save()
+
+        self.api_client.login(username='user3', password='pass')
+        response = self.api_client.get('/api/v1/instance/?lifecycle=production')
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['id'], instance.ref.id)
+
+    @patch_services
+    def test_instance_list_filter_on_lifecycle_sandbox(self, mock_consul, mock_patch_services):
+        """
+        GET - instance list - filter on all sandbox instances.
+        """
+        instance = OpenEdXInstanceFactory(
+            sub_domain='test.com',
+            name='test.com',
+            additional_monitoring_emails=['urgent@opencraft.com']
+        )
+        instance.ref.owner = self.organization2
+        instance.ref.save()
+
+        instance2 = OpenEdXInstanceFactory(sub_domain='test2.com', name='test2.com')
+        instance2.ref.owner = self.organization2
+        instance2.ref.save()
+
+        self.api_client.login(username='user3', password='pass')
+        response = self.api_client.get('/api/v1/instance/?lifecycle=sandbox')
+        self.assertEqual(len(response.data), 1)
+        self.assertEqual(response.data[0]['id'], instance2.ref.id)
+
+
+@ddt.ddt
+@patch(
+    'instance.tests.models.factories.openedx_instance.OpenEdXInstance._write_metadata_to_consul',
+    return_value=(1, True)
+)
```

Comment:
We can probably remove these decorators since we're not using ddt or instance factories.

Reply:
Cool. Thought that was needed for something. Removed.